src/main/scala/vsys/blockchain/state/diffs/CommonValidation.scala

```diff
@@ -24,36 +19,16 @@ object CommonValidation {
 
   def disallowSendingGreaterThanBalance[T <: Transaction](s: StateReader, settings: FunctionalitySettings, blockTime: Long, tx: T): Either[ValidationError, T] =
-
     tx match {
-      case ptx: PaymentTransaction if s.accountPortfolio(EllipticCurve25519Proof.fromBytes(ptx.proofs.proofs.head.bytes.arr).toOption.get.publicKey).balance < (ptx.amount + ptx.fee) =>
-        Left(GenericError(s"Attempt to pay unavailable funds: balance " +
-          s"${s.accountPortfolio(EllipticCurve25519Proof.fromBytes(ptx.proofs.proofs.head.bytes.arr).toOption.get.publicKey).balance} is less than ${ptx.amount + ptx.fee}"))
-      case ttx: TransferTransaction =>
-        val sender: Address = ttx.sender
-
-        val amountDiff = ttx.assetId match {
-          case Some(aid) => Portfolio(0, LeaseInfo.empty, Map(aid -> -ttx.amount))
-          case None => Portfolio(-ttx.amount, LeaseInfo.empty, Map.empty)
-        }
-        val feeDiff = ttx.feeAssetId match {
-          case Some(aid) => Portfolio(0, LeaseInfo.empty, Map(aid -> -ttx.fee))
-          case None => Portfolio(-ttx.fee, LeaseInfo.empty, Map.empty)
-        }
-
-        val accountPortfolio = s.accountPortfolio(sender)
-        val spendings = Monoid.combine(amountDiff, feeDiff)
-
-        lazy val negativeAsset = spendings.assets.find { case (id, amt) => (accountPortfolio.assets.getOrElse(id, 0L) + amt) < 0L }.map { case (id, amt) => (id, accountPortfolio.assets.getOrElse(id, 0L), amt, accountPortfolio.assets.getOrElse(id, 0L) + amt) }
-        lazy val newVSYSBalance = accountPortfolio.balance + spendings.balance
-        lazy val negativeVSYS = newVSYSBalance < 0
-        if (negativeVSYS)
-          Left(GenericError(s"Attempt to transfer unavailable funds:" +
-            s" Transaction application leads to negative vsys balance to (at least) temporary negative state, current balance equals ${accountPortfolio.balance}, spends equals ${spendings.balance}, result is $newVSYSBalance"))
-        else if (negativeAsset.nonEmpty)
-          Left(GenericError(s"Attempt to transfer unavailable funds:" +
-            s" Transaction application leads to negative asset '${negativeAsset.get._1}' balance to (at least) temporary negative state, current balance is ${negativeAsset.get._2}, spends equals ${negativeAsset.get._3}, result is ${negativeAsset.get._4}"))
-        else Right(tx)
+      case ptx: PaymentTransaction =>
+        for {
+          proof <- EllipticCurve25519Proof.fromBytes(ptx.proofs.proofs.head.bytes.arr)
+          sender = proof.publicKey
+          _ <- if(s.accountPortfolio(sender).balance < (ptx.amount + ptx.fee))
+            Left(GenericError(s"Attempt to pay unavailable funds: balance " +
+              s"${s.accountPortfolio(EllipticCurve25519Proof.fromBytes(ptx.proofs.proofs.head.bytes.arr).toOption.get.publicKey).balance} is less than ${ptx.amount + ptx.fee}"))
+            else Right(())
+        } yield tx
```

Comment:
if assert part is removed, it may same as BalanceDiffValidation, however, we can keep it first